// project.proto
// protoc -I=./ --cpp_out=./ ./project.proto

syntax = "proto3";

package Project;



enum MessageType
{
    MT_UNKNOW                         = 0;

    MT_ERROR                          = 1000;
    MT_RANDOM                         = 1001;
    MT_LOGIN                          = 1002;
    MT_LOGIN_RESPONSE                 = 1003;
    MT_LOGOUT                         = 1004;
    MT_LOGOUT_RESPONSE                = 1005;
}

message MessageHeader
{
    MessageType Type = 1;
    string TransmissionID = 2;
}

message UserLogin
{
    MessageHeader header = 1;
    int32 ID = 2;
    string Name = 3;
    string Password = 4;
    string Authority = 5;
    repeated string Nickname = 6;
}

message UserLogout
{
    MessageHeader header = 1;
    int32 ID = 2;
    string Name = 3;
}

message RandomMessage
{
    MessageHeader header = 1;
    int64 ClientSequence = 2;
    int64 ServerSequence = 3;
    int64 SenderProcessID = 4;
    string RandomDescriptor = 5;
}

message ServerError
{
    MessageHeader header = 1;
    int32 ErrorType = 2;
    string ErrorDescriptor = 3;
}
